src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1563,6 +1577,114 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
+    else:
+        function_app_config = {"deployment": None, "runtime": None, "scaleAndConcurrency": None}     
+        
+    if ("deployment" not in function_app_config or function_app_config["deployment"] is None):
+        function_app_config["deployment"] = {"storage": None}
+        
+    deployment_storage = None
+    functionapp_deployment_storage = None
+    if ("storage" not in function_app_config["deployment"] or function_app_config["deployment"]["storage"] is None):
+        function_app_config["deployment"]["storage"] = functionapp_deployment_storage = {"type": "blobContainer", "value": None, "authentication": None}
+        
+    if (functionapp_deployment_storage["value"] is None):
+        if (deployment_storage_name == None):
+            raise ValidationError("Please provide a values for --deployment-storage-name and --deployment-storage-container-name as function app deployment storage value is not set.")
+    
+    if ("authentication" not in functionapp_deployment_storage or functionapp_deployment_storage["authentication"] is None):
+        if (deployment_storage_auth_type == None):
+            raise ValidationError("Please provide a value for --deployment-storage-auth-type as function app deployment storage authentication type is not set.")
+        functionapp_deployment_storage["authentication"] = {"type": "SystemAssignedIdentity", "userAssignedIdentityResourceId": None, "storageAccountConnectionStringName": None}
+
+    # Storage
+    deployment_config_storage_value = None
+    if (deployment_storage_name is not None):
+        deployment_storage = _validate_and_get_deployment_storage(cmd.cli_ctx, resource_group_name, deployment_storage_name)
+        deployment_storage_container = _get_deployment_storage_container(cmd, resource_group_name, deployment_storage_name, deployment_storage_container_name)
+        deployment_storage_container_name = deployment_storage_container.name
+        deployment_config_storage_value = getattr(deployment_storage.primary_endpoints, 'blob') + deployment_storage_container_name
+        functionapp_deployment_storage["value"] = deployment_config_storage_value
+        functionapp_deployment_storage["type"] = "blobContainer"
+        
+    # Authentication
+    assign_identities = None
+    if (deployment_storage_auth_type != None):
+        if deployment_storage_auth_type == 'StorageAccountConnectionString':
+            deployment_storage_conn_string = _get_storage_connection_string(cmd.cli_ctx, deployment_storage)
+            configs = get_site_configs(cmd, resource_group_name, name)
+            
+            if configs.app_settings is None:
+                configs.app_settings = []
+            configs.app_settings.append(NameValuePair(name='DEPLOYMENT_STORAGE_CONNECTION_STRING',
+                                                            value=deployment_storage_conn_string))
+            update_flex_functionapp_configuration(cmd, resource_group_name, name, configs)
```

Comment:
This could just have been 
```suggestion
            update_app_settings(cmd, resource_group_name, name,
                                    ["DEPLOYMENT_STORAGE_CONNECTION_STRING={}".format(deployment_storage_conn_string)])
```

Reply:
done